ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,32 @@ -(void)deleteObject{
     [self remove];
 }
 
+-(void)getExplorerUrlSuccessCallback:(void (^)(NSString*))success errorCallback:(void (^)(NSError*))error {
+    NSLog(@"%@ getExplorerUrl",self.Id);
+    NSMutableString *path = [NSMutableString stringWithFormat:@"https://api.ooni.io/api/v1/measurements?report_id=%@",
+                      self.report_id];
+    if ([self.test_name isEqualToString:@"web_connectivity"])
+        [path appendString:[NSString stringWithFormat:@"&input=%@",
+                            self.url_id.url]];
+    NSURL *url = [NSURL URLWithString:path];
+    NSURLSessionDataTask *downloadTask =
+    [[NSURLSession sharedSession]
+     dataTaskWithURL:url
+     completionHandler:^(NSData *data, NSURLResponse *response, NSError *nserror) {
```

Comment:
Conversationally, you pointed out that AFNetworking is using two callbacks. Maybe I'm missing something, but here above I do actually see that the pattern used is the one where there is a single callback.

Reply:
_Errata corrige_: this is NSURLSession, which is Apple code, rather than AFNetworking.

Reply:
Should we just integrate AFNetworking into the project that is a more robust library?

Reply:
I am missing data points useful to properly reply to that. My tentative response is that we're probably okay with not adding extra libraries, unless NSURL is demonstrably broken, mainly because we don't want extra dependencies for maintenance and code size reasons.

Reply:
Do we agree that we'll reconsider AFNetworking later?

Reply:
sure, better later